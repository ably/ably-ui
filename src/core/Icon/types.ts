import { computedIcons } from "./computed-icons";

export const iconNames = {
  gui: computedIcons.gui,
  display: computedIcons.display,
  social: computedIcons.social,
  other: computedIcons.other,
  tech: computedIcons.tech,
  product: computedIcons.product,
};

export type IconName =
  | (typeof iconNames.gui)[number]
  | (typeof iconNames.display)[number]
  | (typeof iconNames.social)[number]
  | (typeof iconNames.other)[number]
  | (typeof iconNames.tech)[number]
  | (typeof iconNames.product)[number];
