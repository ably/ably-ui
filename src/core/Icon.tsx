import React, { CSSProperties, useCallback, useId, useMemo } from "react";

import {
  defaultIconSecondaryColor,
  getHeroicon,
  setUniqueIds,
  toPascalCase,
} from "./Icon/utils";
import * as IconComponents from "./Icon/components";
import { IconName, IconSize } from "./Icon/types";
import { ColorClass, ColorThemeSet } from "./styles/colors/types";
import { convertTailwindClassToVar } from "./styles/colors/utils";
import cn from "./utils/cn";

export type IconProps = {
  name: IconName;
  size?: IconSize;
  color?: ColorClass | ColorThemeSet;
  secondaryColor?: ColorClass | ColorThemeSet;
  additionalCSS?: string;
  variant?: "outline" | "solid" | "micro" | "mini";
};

// Helper function to get a heroicon component from the static map
const getHeroiconComponent = (
  iconName: string,
): React.ComponentType<React.SVGProps<SVGSVGElement>> | null => {
  // Parse the icon name to extract heroicon name and variant
  // Expected format: icon-gui-{heroicon-name}-{variant}
  const match = iconName.match(/^icon-gui-(.+)-(outline|solid|mini|micro)$/);
  if (!match) {
    return null;
  }

  const [, heroiconName, variant] = match;
  const component = getHeroicon(heroiconName, variant);
  return component;
};

const Icon = ({
  name,
  size = "0.75rem",
  color,
  secondaryColor,
  additionalCSS = "",
  variant,
  ...additionalAttributes
}: IconProps) => {
  const uniqueId = useId();

  const [lightSecondaryColor, darkSecondaryColor] = (
    secondaryColor ?? ""
  ).split(" dark:") as [ColorClass, ColorClass | undefined];

  const iconName =
    variant && !name.endsWith(`-${variant}`) ? `${name}-${variant}` : name;

  // Get the local icon component (memoized)
  const LocalIconComponent = useMemo(():
    | React.ComponentType<React.SVGProps<SVGSVGElement>>
    | undefined => {
    return IconComponents[
      toPascalCase(iconName) as keyof typeof IconComponents
    ];
  }, [iconName]);

  // Get the heroicon component only if no local component is found (memoized)
  const HeroiconComponent = useMemo(() => {
    return LocalIconComponent ? null : getHeroiconComponent(iconName);
  }, [LocalIconComponent, iconName]);

  const setUniqueIdsRef = useCallback(
    (el: SVGSVGElement | null) => {
      setUniqueIds(el, uniqueId);
    },
    [uniqueId],
  );

  const iconSvg = useCallback(
    (secondaryColor?: ColorClass, isDark?: boolean, isThemed?: boolean) => {
      if (LocalIconComponent) {
        let secondaryColorValue;
        if (secondaryColor) {
          secondaryColorValue = convertTailwindClassToVar(secondaryColor);
        } else if (defaultIconSecondaryColor(name)) {
          secondaryColorValue = convertTailwindClassToVar(
            defaultIconSecondaryColor(name),
          );
        }

        return (
          <LocalIconComponent
            className={cn(
              { [`${color}`]: color },
              { [`${additionalCSS}`]: additionalCSS },
              {
                "hidden dark:block": secondaryColor && !isDark && isThemed,
              },
              {
                "dark:hidden": secondaryColor && isDark && isThemed,
              },
            )}
            style={
              {
                width: size,
                height: size,
                ...(secondaryColorValue && {
                  "--ui-icon-secondary-color": secondaryColorValue,
                }),
              } as CSSProperties
            }
            {...additionalAttributes}
            ref={setUniqueIdsRef}
          />
        );
      }

      if (HeroiconComponent) {
        return (
          <HeroiconComponent
            className={cn(
              { [`${color}`]: color },
              { [`${additionalCSS}`]: additionalCSS },
            )}
            style={{
              width: size,
              height: size,
            }}
            {...additionalAttributes}
          />
        );
      }
    },
    [
      LocalIconComponent,
      HeroiconComponent,
      iconName,
      size,
      color,
      additionalCSS,
      lightSecondaryColor,
      darkSecondaryColor,
      name,
    ],
  );

  if (lightSecondaryColor && darkSecondaryColor) {
    return (
      <>
        {iconSvg(lightSecondaryColor, false, !!darkSecondaryColor)}
        {iconSvg(darkSecondaryColor, true, true)}
      </>
    );
  } else if (lightSecondaryColor) {
    return iconSvg(lightSecondaryColor, false, !!darkSecondaryColor);
  } else {
    return iconSvg();
  }
};

export default Icon;
