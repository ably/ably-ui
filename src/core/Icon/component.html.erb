<%= tag.svg class: "#{color} #{additional_css}", style: "width: #{size}; height: #{size}", **additional_attributes do %>
  <%= tag.use "xlink:href": "#sprite-#{name}" %>
<% end %>
