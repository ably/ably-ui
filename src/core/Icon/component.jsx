import React from "react";
import T from "prop-types";

const Icon = ({ name, size = "0.75rem", color = "", additionalCSS = "", ...additionalAttributes }) => {
  return (
    <svg className={`${color} ${additionalCSS}`} style={{ width: size, height: size }} {...additionalAttributes}>
      <use xlinkHref={`#sprite-${name}`} />
    </svg>
  );
};

Icon.propTypes = {
  name: T.string.isRequired,
  size: T.string,
  color: T.string,
  additionalCSS: T.string,
};

export default Icon;
