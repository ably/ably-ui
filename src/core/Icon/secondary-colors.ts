import { ColorClass } from "../styles/colors/types";
import { IconName } from "./types";

export const defaultIconSecondaryColor = (name: IconName) => {
  const colorLookup = {
    "icon-gui-check-circled-fill": "text-white",
  } as Record<IconName, ColorClass>;

  return colorLookup[name];
};
