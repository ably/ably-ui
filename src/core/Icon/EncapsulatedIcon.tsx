import React from "react";
import Icon, { IconProps } from "../Icon";
import useTheming from "../hooks/useTheming";
import { Theme } from "../styles/colors/types";
import { IconSize } from "./types";

type EncapsulatedIconProps = {
  theme?: Theme;
  className?: string;
  innerClassName?: string;
  iconSize?: IconSize;
} & IconProps;

const ICON_SIZE_REDUCTION = 12;
const ICON_HEIGHT_REDUCTION = 2;

const EncapsulatedIcon = ({
  theme = "dark",
  size = "40px",
  iconSize,
  className,
  innerClassName,
  ...iconProps
}: EncapsulatedIconProps) => {
  const { themeColor } = useTheming({
    baseTheme: "dark",
    theme,
  });
  let computedIconSize = size,
    computedIconHeight = size;

  if (iconSize) {
    computedIconSize = iconSize;
  } else if (size.toString().endsWith("px")) {
    const numericalSize = parseInt(size, 10);
    computedIconSize = `${numericalSize - ICON_SIZE_REDUCTION}px`;
    computedIconHeight = `${numericalSize - ICON_HEIGHT_REDUCTION}px`;
  } else {
    console.warn(
      `EncapsulatedIcon: Non-pixel units might not behave as expected`,
    );
  }

  return (
    <div
      className={`p-1 rounded-lg ${theme === "light" ? "bg-gradient-to-t" : "bg-gradient-to-b"} ${themeColor("from-neutral-900")} ${className ?? ""}`}
      style={{ width: size, height: size }}
    >
      <div
        className={`flex items-center justify-center rounded-lg ${themeColor("bg-neutral-1100")} ${innerClassName ?? ""}`}
        style={{ height: computedIconHeight }}
      >
        <Icon size={computedIconSize} {...iconProps} />
      </div>
    </div>
  );
};

export default EncapsulatedIcon;
