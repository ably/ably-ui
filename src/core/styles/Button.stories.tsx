import React from "react";
import Icon from "../Icon";

export default {
  title: "CSS/Button",
};

export const Standard = {
  render: () => (
    <div className="flex flex-col gap-16 max-w-[300px]">
      <button type="button" className="ui-btn">
        Primary button
      </button>
      <button type="button" className="ui-btn-alt">
        Primary button alternative
      </button>
      <button type="button" className="ui-btn-secondary">
        Secondary button
      </button>
      <button type="button" className="ui-btn" disabled>
        Unavailable primary button
      </button>
      <button type="button" className="ui-btn-alt" disabled>
        Unavailable primary button alternative
      </button>
      <button type="button" className="ui-btn-secondary" disabled>
        Unavailable secondary button
      </button>
    </div>
  ),
};

export const Inverted = {
  render: () => (
    <div className="flex flex-col gap-16 max-w-[300px]">
      <button type="button" className="ui-btn-invert">
        Primary button
      </button>
      <button type="button" className="ui-btn-secondary-invert">
        Secondary button
      </button>
      <button type="button" className="ui-btn-invert" disabled>
        Unavailable primary button
      </button>
      <button type="button" className="ui-btn-secondary-invert" disabled>
        Unavailable secondary button
      </button>
    </div>
  ),
};

export const WithIcons = {
  render: () => (
    <div className="flex flex-col gap-16 max-w-[300px]">
      <button type="button" className="ui-btn">
        <Icon name="icon-gui-search" size="1rem" additionalCSS="ui-btn-icon" />
        Primary button with icon
      </button>
      <button type="button" className="ui-btn-alt">
        <Icon name="icon-gui-search" size="1rem" additionalCSS="ui-btn-icon" />
        Primary button alternative with icon
      </button>
      <button type="button" className="ui-btn-secondary">
        <Icon name="icon-gui-search" size="1rem" additionalCSS="ui-btn-icon" />
        Secondary button with icon
      </button>
      <button type="button" className="ui-btn" disabled>
        <Icon name="icon-gui-search" size="1rem" additionalCSS="ui-btn-icon" />
        Unavailable primary button with icon
      </button>
      <button type="button" className="ui-btn-alt" disabled>
        <Icon name="icon-gui-search" size="1rem" additionalCSS="ui-btn-icon" />
        Unavailable primary button alternative with icon
      </button>
      <button type="button" className="ui-btn-secondary" disabled>
        <Icon name="icon-gui-search" size="1rem" additionalCSS="ui-btn-icon" />
        Unavailable secondary button with icon
      </button>
    </div>
  ),
};

export const LinkStyledAsButton = {
  render: () => (
    <div className="flex flex-col gap-16 max-w-[300px]">
      <a href="#" role="button" className="ui-btn">
        Link
      </a>
      <a href="#" role="button" className="ui-btn ui-btn-disabled">
        Disabled link
      </a>
    </div>
  ),
};

export const ExtraSmall = {
  render: () => (
    <div className="flex flex-col gap-16 max-w-[300px]">
      <button type="button" className="ui-btn p-btn-xsmall text-btn4">
        Primary button
      </button>
      <button type="button" className="ui-btn p-btn-xsmall text-btn4">
        <Icon
          name="icon-gui-search"
          size="1rem"
          additionalCSS="ui-btn-icon-xsmall"
        />
        Primary button with icon
      </button>
      <button type="button" className="ui-btn-alt p-btn-xsmall text-btn4">
        Primary button alternative
      </button>
      <button type="button" className="ui-btn-secondary p-btn-xsmall text-btn4">
        Secondary button
      </button>
    </div>
  ),
};

export const Small = {
  render: () => (
    <div className="flex flex-col gap-16 max-w-[300px]">
      <button type="button" className="ui-btn p-btn-small text-btn3">
        Primary button
      </button>
      <button type="button" className="ui-btn p-btn-small text-btn3">
        <Icon
          name="icon-gui-search"
          size="1rem"
          additionalCSS="ui-btn-icon-small"
        />
        Primary button with icon
      </button>
      <button type="button" className="ui-btn-alt p-btn-small text-btn3">
        Primary button alternative
      </button>
      <button type="button" className="ui-btn-secondary p-btn-small text-btn3">
        Secondary button
      </button>
    </div>
  ),
};

export const Large = {
  render: () => (
    <div className="flex flex-col gap-16 max-w-[300px]">
      <button type="button" className="ui-btn p-btn-large text-btn1">
        Primary button
      </button>
      <button type="button" className="ui-btn p-btn-large text-btn1">
        <Icon name="icon-gui-search" size="1rem" additionalCSS="ui-btn-icon" />
        Primary button with icon
      </button>
      <button type="button" className="ui-btn-alt p-btn-large text-btn1">
        Primary button alternative
      </button>
      <button type="button" className="ui-btn-secondary p-btn-large text-btn1">
        Secondary button
      </button>
    </div>
  ),
};
