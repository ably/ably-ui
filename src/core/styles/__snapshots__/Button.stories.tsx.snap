// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button Primary smoke-test 1`] = `
<div class="grid grid-cols-2 sm:grid-cols-4 gap-16">
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-lg"
      >
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary"
      >
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-sm"
      >
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-xs"
      >
        Extra small
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-lg ui-button-lg-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary ui-button-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-sm ui-button-sm-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-xs"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Extra small
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-lg ui-button-lg-right-icon"
      >
        Large
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary ui-button-right-icon"
      >
        Regular
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-sm ui-button-sm-right-icon"
      >
        Small
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-xs"
      >
        Extra small
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-lg"
              disabled
      >
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary"
              disabled
      >
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-sm"
              disabled
      >
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-primary-xs"
              disabled
      >
        Extra small
      </button>
    </div>
  </div>
</div>
`;

exports[`Components/Button Priority smoke-test 1`] = `
<div class="grid grid-cols-2 sm:grid-cols-4 gap-16">
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-lg"
      >
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority"
      >
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-sm"
      >
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-xs"
      >
        Extra small
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-lg ui-button-lg-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority ui-button-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-sm ui-button-sm-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-xs"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Extra small
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-lg ui-button-lg-right-icon"
      >
        Large
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority ui-button-right-icon"
      >
        Regular
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-sm ui-button-sm-right-icon"
      >
        Small
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-xs"
      >
        Extra small
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-lg"
              disabled
      >
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority"
              disabled
      >
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-sm"
              disabled
      >
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-priority-xs"
              disabled
      >
        Extra small
      </button>
    </div>
  </div>
</div>
`;

exports[`Components/Button ReactComponent smoke-test 1`] = `
<div class="grid sm:grid-cols-3 gap-16">
  <div>
    <button class="ui-button-priority-lg">
      Hello React!
    </button>
  </div>
  <div>
    <button class="ui-button-primary ui-button-left-icon">
      <svg class
           style="width: 0.75rem; height: 0.75rem;"
      >
        <use xlink:href="#sprite-icon-gui-magnifying-glass-outline">
        </use>
      </svg>
      Hello React!
    </button>
  </div>
  <div>
    <button class="ui-button-secondary-xs">
      Hello React!
      <svg class
           style="width: 0.75rem; height: 0.75rem;"
      >
        <use xlink:href="#sprite-icon-gui-arrow-long-right-outline">
        </use>
      </svg>
    </button>
  </div>
</div>
`;

exports[`Components/Button Secondary smoke-test 1`] = `
<div class="grid grid-cols-2 sm:grid-cols-4 gap-16">
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-lg"
      >
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary"
      >
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-sm"
      >
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-xs"
      >
        Extra small
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-lg ui-button-lg-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary ui-button-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-sm ui-button-sm-left-icon"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-xs"
      >
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-search">
          </use>
        </svg>
        Extra small
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-lg ui-button-lg-right-icon"
      >
        Large
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary ui-button-right-icon"
      >
        Regular
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-sm ui-button-sm-right-icon"
      >
        Small
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-xs"
      >
        Extra small
        <svg style="width: 1.5rem; height: 1.5rem;">
          <use xlink:href="#sprite-icon-gui-arrow-right">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="flex flex-col gap-16">
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-lg"
              disabled
      >
        Large
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary"
              disabled
      >
        Regular
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-sm"
              disabled
      >
        Small
      </button>
    </div>
    <div class="inline-flex ml-8 ">
      <button type="button"
              aria-describedby="tooltip"
              class="p-0 relative focus:outline-none h-[1rem] h-[60px]"
      >
      </button>
      <button type="button"
              class="ui-button-secondary-xs"
              disabled
      >
        Extra small
      </button>
    </div>
  </div>
</div>
`;
